modin/data_management/functions/groupby_function.py

```diff
@@ -301,19 +365,41 @@ def caller(
             reduce_func=reduce_func,
             reduce_args=reduce_args,
             drop=drop,
+            selection=selection,
             method=method,
         )
 
         # If `by` is a ModinFrame, then its partitions will be broadcasted to every
         # `self` partition in a way determined by engine (modin_frame.groupby_reduce)
         # Otherwise `by` was already bound to the Map function in `build_map_reduce_functions`.
         broadcastable_by = getattr(by, "_modin_frame", None)
-        apply_indices = list(map_func.keys()) if isinstance(map_func, dict) else None
+        apply_indices = None
+        if isinstance(map_func, dict):
+            apply_indices = tuple(map_func.keys())
+        if selection is not None and apply_indices is None:
+            apply_indices = selection if is_list_like(selection) else (selection,)
         new_modin_frame = qc._modin_frame.groupby_reduce(
-            axis, broadcastable_by, map_fn, reduce_fn, apply_indices=apply_indices
+            axis, broadcastable_by, map_fn, reduce_fn, selection=apply_indices
         )
 
         result = query_compiler.__constructor__(new_modin_frame)
+        if len(result.columns) == 0 and len(query_compiler.columns) != 0:
```

Comment:
the full duplication of [this](https://github.com/modin-project/modin/pull/3298/files#diff-a7c5b49b77f6301421bad047e3b0b1cfa30f0f8b08242e59145359e7f0d07748R2856), should be moved to the front-end in the future

Reply:
moved this logic to the [`_verify_aggregation_result`](https://github.com/YarShev/modin/blob/9be84a9f8deaf70819db5b437776441fbb971d5c/modin/pandas/groupby.py#L1154) method, located at the front-end